src/pages/InfiniteScrollTest/index.tsx

```diff
@@ -37,22 +39,20 @@ export default function InfiniteScrollTest() : JSX.Element {
   async function ReturnUsers(
     lastElement : User | null, pageNumber : number, pageSize : number,
   ) : Promise<Array<User>> {
-    let firestoreQuery;
+    let query;
 
     if (pageNumber === 1 || lastElement == null) {
-      firestoreQuery = firestore()
-        .collection('users')
+      query = userAPI.userFirestoreCollection()
         .orderBy('username')
         .limit(pageSize);
     } else {
-      firestoreQuery = firestore()
-        .collection('users')
+      query = userAPI.userFirestoreCollection()
```

Comment:
I would suggest have a paginated query in the API like `Users.all(limit=pageSize, startAfter=lastElement.userName)`, by default sorting by username, and all those parameters optional

Reply:
Fixed by new commit.